pymeasure/instruments/hp/hp3478A.py

```diff
@@ -522,9 +522,8 @@ def write_calibration_data(self, cal_data, verify_calibration_data=True):
         'verify_calibration_data' to False.
 
         """
-        if verify_calibration_data:
-            if not self.verify_calibration_data(cal_data):
-                raise Exception("cal_data verification fail.")
+        if verify_calibration_data and not self.verify_calibration_data(cal_data):
+            raise Exception("cal_data verification fail.")
```

Comment:
As far as I know, you should not use 'Exception', but some subclass. Here I'd use a Value Error (use it in the tests as well).

Reply:
Done.